src/Altinn.App.Api/Controllers/LookupOrganisationController.cs

```diff
@@ -43,11 +44,39 @@ ILogger<LookupOrganisationController> logger
     [HttpGet]
     [Route("{orgNr}")]
     [ProducesResponseType(typeof(LookupOrganisationResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
     public async Task<ActionResult<LookupOrganisationResponse>> LookUpOrganisation([FromRoute] string orgNr)
     {
         _logger.LogInformation($"Looking up organisation with orgNr: {orgNr}");
-        Organization? organisation = await _organisationClient.GetOrganization(orgNr);
 
-        return Ok(LookupOrganisationResponse.CreateFromOrganisation(organisation));
+        var organisationResult = await GetOrganisationDataOrError(orgNr);
+        if (!organisationResult.Success)
+        {
+            ProblemDetails problemDetails = organisationResult.Error;
+            return StatusCode(problemDetails.Status ?? 500, problemDetails);
+        }
+
+        return Ok(LookupOrganisationResponse.CreateFromOrganisation(organisationResult.Ok));
+    }
+
+    private async Task<ServiceResult<Organization?, ProblemDetails>> GetOrganisationDataOrError(string orgNr)
+    {
+        Organization? organisation;
+        try
+        {
+            organisation = await _organisationClient.GetOrganization(orgNr);
+        }
+        catch
+        {
+            return new ProblemDetails
```

Comment:
Should the exception be logged somewhere?

Reply:
log and trace coming up